### PR TITLE
use pubsub for making build requests

### PR DESCRIPTION
**Pull request by GGAlanSmithee**
@brokenprogrammer 

This replaces custom build protocol with pubusb and creates classes for different types of peers. It also:

* restructures the code a bit
* removes unused code
* adds the `ClientPeer` and `BuilderPeer` types
* adds an additional program entry point for the builder peer, since those are kind of distinct (atleast for the moment, we might not want it to be though)

Please have a look at it! Thanks

### CODE REVIEW

**Comment by GGAlanSmithee** [comment]
@brokenprogrammer so the pubsub is a quite interesting alternative to the custom protocols that libp2p offers. You can read about it (and the specific implementation GossipSub) here: https://docs.libp2p.io/concepts/publish-subscribe/

I am not 100% sure about the tradeoffs between protocols and pubusb, because it seems to me that pubsub is a direct replacement and a better alternative, but if that is the case, it makes no sense to even have protocols, so it's probably not. But will have to read up on it a bit more.

**Comment by GGAlanSmithee** [comment]
With this merged, any connected builder will receive a request every 10 secs looking like:

image

still just a mockup ATM of course

EDIT: LOL, blurry image. Wonder if it's the windows snip tool or github downsamples it or something, wth..

File: README.md [hunk 1]
```diff
@@ -20,20 +20,20 @@ Work In Progress **NOT** ready for use
 
 ## Run instructions
 
-`npm start`
+1. `npm run start:client` _runs the client peer_
+1. `npm run start:builder` _runs the builder peer_
```
Comment (brokenprogrammer) [inline README.md:24]: Doesn't really matter but should prolly be `2.` here

**Review by brokenprogrammer — APPROVED** [review]
Looks good, code is easy to follow and grasp! Nice work

**Comment by GGAlanSmithee** [comment]
Thanks @brokenprogrammer ! merging

**Pull request merged**